core/src/util/vboMesh.cpp

```diff
@@ -68,15 +68,18 @@ void VboMesh::addVertex(GLbyte* _vertex) {
 void VboMesh::addVertices(GLbyte* _vertices, int _nVertices) {
 
     if (m_isUploaded) {
-        logMsg("%s\n", "VboMesh cannot add vertices after upload!");
+        logMsg("ERROR: VboMesh cannot add vertices after upload!\n");
         return;
     }
     
     // Only add up to 65535 vertices, any more will overflow our 16-bit indices
     int indexSpace = MAX_INDEX_VALUE - m_nVertices;
-    if (_nVertices > indexSpace) {
-        _nVertices = indexSpace;
-        logMsg("WARNING: Tried to add more vertices than available in index space\n");
+    if (_nVertices > indexSpace || m_backupMesh) {
```

Comment:
If ever we got to a dataset with any iteration of addVertices adding vertices more than "indexSpace" then it will be a infinite loop of m_backupMeshes. We should have a catch for that.